src/content/doc-surrealql/datamodel/ids.mdx

```diff
@@ -8,209 +8,552 @@ description: In SurrealDB, document record IDs store both the table name, and th
-CREATE temperature:rand() SET time = time::now(), celsius = 37.5;
+DEFINE FIELD possessions ON TABLE person TYPE option<array<record>>;
+DEFINE FIELD friends ON TABLE person TYPE option<array<record<person>>>;
 
-// Generate a ULID-based record ID
-CREATE temperature:ulid() SET time = time::now(), celsius = 37.5;
-// Generate a UUIDv7-based record ID
-CREATE temperature:uuid() SET time = time::now(), celsius = 37.5;
+CREATE person SET
+    possessions = [ book:one, house:one],
+    friends = [ person:one, person:two ];
+```
+
+Be sure to use just `record` instead of `record<any>`, as `<any>` here would imply actual records of a table called `any`.
+
+```surql
+DEFINE FIELD possessions ON TABLE person TYPE option<array<record<any>>>;
+
+-- Won't work, 'book' and 'house' are not of table 'any'
+CREATE person SET
+    possessions = [ book:one, house:one ];
+
+-- Actually expects this, which is probably
+-- not what the DEFINE FIELD intended
+CREATE person SET
+    possessions = [ any:one, any:two ];
 ```
 
-### Record ranges
-SurrealDB supports the ability to query a range of records, using the record ID. The record ID ranges, retrieve records using the natural sorting order of the record IDs. These range queries can be used to query a range of records in a timeseries context.
+## Record ranges
+
+SurrealDB supports the ability to query a range of records, using the record ID. Record ID range queries retrieve records using the natural sorting order of the record IDs, making a table scan unnecessary. These range queries can be used to query a range of records in a timeseries context.
 
 ```surql
 -- Select all person records with IDs between the given range
 SELECT * FROM person:1..1000;
+
 -- Select all records for a particular location, inclusive
-SELECT * FROM temperature:['London', NONE]..=['London', time::now()];
+SELECT * FROM temperature:['London', NONE]..=['London', ..];
+
 -- Select all temperature records with IDs less than a maximum value
 SELECT * FROM temperature:..['London', '2022-08-29T08:09:31'];
+
 -- Select all temperature records with IDs greater than a minimum value
 SELECT * FROM temperature:['London', '2022-08-29T08:03:39']..;
+
 -- Select all temperature records with IDs between the specified range
 SELECT * FROM temperature:['London', '2022-08-29T08:03:39']..['London', '2022-08-29T08:09:31'];
 ```
 
-## Field names
+The following example shows the difference in performance between a regular query that uses a `WHERE` clause and a record range scan.
 
-### Valid field names
+```surql
+FOR $num IN 0..=100000 {
+  CREATE person SET id = $num, num = $num  
+};
 
-Similar to record IDs, field names can be constructed from ASCII characters, underscores, and numbers:
+-- Assign the output to an unused parameter
+-- to avoid excessive output
+LET $_ = SELECT * FROM person WHERE num IN 0..=1000;
+LET $_ = SELECT * FROM person:0..=1000;
+```
+
+## Record IDs are links
+
+As a record ID 
+
+/docs/surrealql/datamodel/records
```

Comment:
? @Dhghomon

Reply:
? indeed! Decided against adding that section, forgot to remove it. https://github.com/surrealdb/docs.surrealdb.com/pull/1170/commits/2a87bc360e750049f0dffe4c32f5d076d4b6695c